src/ui/preferences.cc

```diff
@@ -174,6 +174,11 @@ Preferences::Preferences( QWidget * parent, Config::Class & cfg_ ):
   ui.hideSingleTab->setChecked( p.hideSingleTab );
   ui.mruTabOrder->setChecked( p.mruTabOrder );
   ui.enableTrayIcon->setChecked( p.enableTrayIcon );
+
+#ifdef Q_OS_MACOS // macOS uses the dock menu instead of the tray icon
+  ui.enableTrayIcon->hide();
+#endif
+
   ui.startToTray->setChecked( p.startToTray );
   ui.closeToTray->setChecked( p.closeToTray );
```

Comment:
p.closeToTray was not saved on macos,

Reply:
It will forever be false on macOS
https://github.com/xiaoyifang/goldendict-ng/pull/1935/files#diff-d975ba6a49f5eefe4eaa02271dd5c75cf9ac9a9391e204395e64722659178250R345-R346

Reply:
Can the users see&set this option on preference?

Reply:
this option should also not visible on macos?

Reply:
The entire `enableTrayIcon` group, which contains 3 options, is not visible.
<img width="500" src="https://github.com/user-attachments/assets/ae516749-8ba7-4a99-a4d6-97c3b8c9c756">

Reply:
The entire `enableTrayIcon` group, which contains 3 options, is not visible.

<img width="400" src="https://github.com/user-attachments/assets/a4c9d6af-dcd0-4fb4-99b8-52d6dbe09bcd">